server/src/server.ts

```diff
@@ -94,9 +95,9 @@ connection.onShutdown(() => {
 })
 
 connection.onDidChangeConfiguration((change) => {
-  logger.level = change.settings.bitbake.loggingLevel
-  parseOnSave = change.settings.bitbake.parseOnSave
-  const bitbakeFolder = expandSettingPath(change.settings.bitbake.pathToBitbakeFolder, { workspaceFolder })
+  logger.level = change.settings.bitbake?.loggingLevel ?? logger.level
+  parseOnSave = change.settings.bitbake?.parseOnSave ?? parseOnSave

```

Comment:
VSCode guarantees default values for these. I guess it's for nvim which doesn't have them? This looks great however it's not related to the current commit message. In order to keep the git history more readable, I suggest splitting this into a separate commit: `Fix: Keep server settings in case a change is not defined`

Reply:
Fixed